apps/desktop-wallet/src/features/assetsLists/NFTsGrid.tsx

```diff
@@ -27,46 +27,65 @@ import NFTCard from '@/components/NFTCard'
 import SkeletonLoader from '@/components/SkeletonLoader'
 import { TableRow } from '@/components/Table'
 import ExpandRowButton from '@/features/assetsLists/ExpandRowButton'
+import PlaceholderText from '@/features/assetsLists/PlaceholderText'
 import { AddressTokensTabsProps, TokensTabsBaseProps } from '@/features/assetsLists/types'
 import { deviceBreakPoints } from '@/style/globalStyles'
 import { TokenId } from '@/types/tokens'
 
 export const AddressNFTsGrid = ({ addressHash, ...props }: AddressTokensTabsProps) => {
+  const { t } = useTranslation()
   const {
     data: { nftIds },
     isLoading
   } = useFetchAddressTokensByType({ addressHash, includeAlph: false })
 
-  return <NFTsGrid {...props} columns={4} nftIds={nftIds} isLoading={isLoading} />
+  return (
+    <NFTsGrid
+      {...props}
+      columns={4}
+      nftIds={nftIds}
+      isLoading={isLoading}
+      placeholderText={t("This address doesn't have any NFTs.")}
+    />
+  )
 }
 
 export const WalletNFTsGrid = (props: TokensTabsBaseProps) => {
+  const { t } = useTranslation()
   const {
     data: { nftIds },
     isLoading
   } = useFetchWalletTokensByType({ includeAlph: false })
 
-  return <NFTsGrid {...props} columns={5} nftIds={nftIds} isLoading={isLoading} />
+  return (
+    <NFTsGrid
+      {...props}
+      columns={5}
+      nftIds={nftIds}
+      isLoading={isLoading}
+      placeholderText={t("The wallet doesn't have any NFTs. NFTs of all your addresses will appear here.")}
+    />
+  )
 }
 
 interface NFTsGridProps extends TokensTabsBaseProps {
   columns: number
   nftIds: TokenId[]
   isLoading: boolean
+  placeholderText: string
 }
 
-const NFTsGrid = ({ className, isExpanded, onExpand, columns, nftIds, isLoading }: NFTsGridProps) => (
+const NFTsGrid = ({ className, isExpanded, onExpand, columns, nftIds, isLoading, placeholderText }: NFTsGridProps) => (
   <>
     <motion.div {...fadeIn} className={className}>
-      <Grid role="row" tabIndex={isExpanded ? 0 : -1} columns={columns}>
-        {isLoading ? (
-          <NFTsLoader />
-        ) : nftIds.length === 0 ? (
-          <NoNFTsPlaceholder />
-        ) : (
-          nftIds.map((nftId) => <NFTCard key={nftId} nftId={nftId} />)
-        )}
-      </Grid>
+      {!isLoading && nftIds.length === 0 && <PlaceholderText>{placeholderText}</PlaceholderText>}
+
+      {isLoading ||
+        (nftIds.length > 0 && (
+          <Grid role="row" tabIndex={isExpanded ? 0 : -1} columns={columns}>
+            {isLoading ? <NFTsLoader /> : nftIds.map((nftId) => <NFTCard key={nftId} nftId={nftId} />)}
+          </Grid>
+        ))}
```

Comment:
We're inside the `<motion.div`. How'd you go about it?

Reply:
I'd extract everything to a `renderGrid` function. 
But that's a nit, I'm fine with keeping that as it is now.

Reply:
I see. I am not a big fan of breaking the render flow with render functions so I'll leave it like this since it's a nit.

Let me know if this PR needs anything else from my side!